src/models/thread.ts

```diff
@@ -363,7 +369,7 @@ export class Thread extends ReadReceipt<ThreadEmittedEvents, ThreadEventHandlerM
      * to the start (and not the end) of the timeline.
      * @param emit - whether to emit the Update event if the thread was updated or not.
      */
-    public async addEvent(event: MatrixEvent, toStartOfTimeline: boolean, emit = true): Promise<void> {
+    public addEvent(event: MatrixEvent, toStartOfTimeline: boolean, emit = true): void {
```

Comment:
This was unnecessarily async. I fixed it but that set sonarcloud off on the function complexity, so I had to pull a chunk of it out to reduce the complexity. I can pull this out to a separate PR if necessary but hopefully it's obvious enough what's going on.

Reply:
ah okay, good to know. I assume dit wouldn't let me do that with code complexity.